src/components/app/index.tsx

```diff
@@ -105,38 +136,55 @@ export default class App extends Component<Props, State> {
   }
 
   async updateImage(index: number) {
-    const { sourceData, images } = this.state;
+    const { sourceData, sourceFile, images } = this.state;
     if (!sourceData) return;
     let image = images[index];
     // Each time we trigger an async encode, the ID changes.
     const id = ++image.counter;
     image.loading = true;
     this.setState({ });
-    let result = await this.updateCompressedImage(sourceData, image.type, image.options);
+    if(image.type == 'original' && sourceFile) {
+      image.extension = sourceFile.name.split('.').pop() || "";;
+      // If we don't use this then image data is a bitmap from identity encoder.
+      image.dataUrl = URL.createObjectURL(new Blob([sourceFile]));
+    }
+    let compressedImage = await this.updateCompressedImage(sourceData, image.type, image.options);
+    let imageBitmap = await createImageBitmap(compressedImage.data);
     image = this.state.images[index];
     // If another encode has been intiated since we started, ignore this one.
     if (image.counter !== id) return;
-    image.data = result;
+    image.data = imageBitmap;
+    if(image.type != 'original') {
+      if(compressedImage.data instanceof Blob) {
+        image.dataUrl = URL.createObjectURL(compressedImage.data);
+      }
+      else{
```

Comment:
`else` on previous line and space before and after.

Reply:
mate - not sure how I missed any of these...

Reply:
The spectator sees more of the game 😀